obs-visca-control.lua

```diff
@@ -593,18 +660,37 @@ end
 function signal_on_activate(calldata)
     local source = obs.calldata_source(calldata, "source")
 	local settings = obs.obs_source_get_settings(source)
+	local source_name = obs.obs_source_get_name(source)
 
-    local do_preset = false
+    local do_action = false
     local active = obs.obs_data_get_int(settings, "scene_active")
     if (active == action_active.Program) or (active == action_active.Always) then
-        do_preset = true
+        do_action = true
     end
 
-    if do_preset then
-        do_cam_scene_action(settings)
+    if do_action then
+        do_cam_scene_action(settings, source_name)
     end
 
 	obs.obs_data_release(settings)
 end
 
+function signal_on_deactivate(calldata)
+    local source = obs.calldata_source(calldata, "source")
+	local source_name = obs.obs_source_get_name(source)
+    for a, data in pairs(previous_actions) do
+        -- log("pair %d action %s %s", a, data.source_name or 0, source_name)
+        if data.source_name == source_name and data.scene_action == actions.Animate then
```

Comment:
This is a valid and correct check that is missing in `fe_callback`. The limitation to `actions.Animate` actions only is debatable. For now it's ok, but you can probably leave it out.